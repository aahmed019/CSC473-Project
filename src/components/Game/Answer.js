import React, { Component } from 'react';
import games from './games.json';

class Answer extends Component {
    constructor(props) {
        super(props);
        this.state = {
            gameIndex: this.props.id,
            questionIndex: this.props.qId,
        };
        this.enterNum = this.enterNum.bind(this);
        this.enterText = this.enterText.bind(this);
    }
    //this function will add value of button pressed to pound value
    enterNum(e) {
        document.getElementById("pound").value += e.currentTarget.value;
        console.log("pound value", document.getElementById("pound").value)
    }
    //this enter will get current value of the input and assign that value to submit button
    enterText() {
        document.getElementById("submitBttn").value = document.getElementById("answer").value;
    }

    componentDidUpdate(prevProps) {
        if (prevProps.qId !== this.props.qId) {
            this.setState({
                questionIndex: this.props.qId,  //update question index to re-render
            })
        }
    }

    render() {
        if (games[this.state.gameIndex].Answer_Type[this.state.questionIndex] == "number") {
            return (
                // Construct Num Pad
                <div key={this.props.qId} id="numPad">
                    <button type="button" className="number" onClick={this.enterNum} value="7">7</button>
                    <button type="button" className="number" onClick={this.enterNum} value="8">8</button>
                    <button type="button" className="number" onClick={this.enterNum} value="9">9</button>

                    <button type="button" className="number" onClick={this.enterNum} value="4">4</button>
                    <button type="button" className="number" onClick={this.enterNum} value="5">5</button>
                    <button type="button" className="number" onClick={this.enterNum} value="6">6</button>

                    <button type="button" className="number" onClick={this.enterNum} value="1">1</button>
                    <button type="button" className="number" onClick={this.enterNum} value="2">2</button>
                    <button type="button" className="number" onClick={this.enterNum} value="3">3</button>

                    <button type="button" className="number" onClick={this.enterNum} value=".">.</button>
                    <button type="button" className="number" onClick={this.enterNum} value="0">0</button>
                    <button type="button" className="number" id="pound" onClick={this.props.action} value="">#</button>
                </div>
            )
        }
        else {
            return (
                <div key={this.props.qId} id="textAnswer">
                    <input id="answerBox" type="text" className="text-center textbox" onChange={this.enterText} />
                    <br />
                    <button id="submitBttn" className="btn-large  btn-success" type="button" onClick={this.props.action} value="">&nbsp; Submit &nbsp;</button>
                </div>
            )

        }

    }
}

export default Answer;