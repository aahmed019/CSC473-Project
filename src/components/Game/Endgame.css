.ending-page {
    background-image: url(../home/home-background.jpg);
    font-family: "Courier New";
    min-height: 100vh;
    color: white;
}

#message {
    text-align: center;
    font-size: 85px;
    color: white;
    max-width: 80%;
    margin: 0 auto;
}

#outcome-img {
    flex: center;
    max-height: 400px;
    max-width: 500px;
    margin-top: 30px;
    margin-bottom: 30px;
}

#review-form {
    margin: 0 auto;
}

.form-label {
    font-size: 50px;
}

#rating-select{
    font-size: 30px;
}
#feedback{
    font-size: 30px;
}

.review-submit-button {
    background-color: rgb(255, 187, 0);
}

.section-divider p {
    font-size: 30px;
}