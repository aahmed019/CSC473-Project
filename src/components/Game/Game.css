
.textbox{
  width:70%;
  font-size:30px;
  text-align: center;
}
    
.g-image{

  align-content: center;
  margin-top:10%;
  margin-left: 15%;
  width: 30vw;
  height: 25vh;
}


.btn-lg {
  font-size: 3vh;
  color: white;
}
#hintBttn{
  font-size:3vh;
}

.exit{
  /* margin-top: 5%;
  margin-left: 5%; */
  margin: 2vh 2vh 2vh 2vh;
  text-align: left;

}

.exit a{
  text-decoration: none; 
  color: white;
  font-size: 3vh;
}

.start {
  margin: 2vh 2vh 2vh 2vh;
}

.hint{
  font-size:200%;
}
.questN{
  font-size: 200%;
  font-style:italic ;
}
.quest{
  font-size: 225%;
  text-align: center;
  width: auto;
  padding: 75px;
  
}

.toggleGame-button-line {
  width: 2vh;
  height: 0.25vh;
  background: black;
}
.spaceGame-btw-logo-items {
  flex: .95;
}
.gameTitle{
  text-decoration: underline;
}
.imgG{
  width: 85%;
  height: auto;
}

.gameButton{
  font-size:250%
}

/* THIS IS FOR GAME LIST */
.game-list{
  display: grid;
  width: 50%;
  margin-left: 3.5%;
}

.card{
  list-style-type: none;
}

.card-img-top{
  height: 30vh;
  width: 100%;
}

.card-text{
  font-size: 200%;
  text-align: left;
}

/* THIS IS FOR QUESTION DISPLAY */
.imgG{
  align-content: center;
  margin-top:10%;
  margin-left: 15%;
  margin-right: 15%;
  width: 60%;
  height: 25vh;
}

/* THIS IS FOR NUMBER ANSWER STYLE */
#numPad{
  display: grid;
  grid-template-columns: repeat(3,1fr);
  grid-gap: 10px;
  width: 50%;
  margin-left: 25%;
  padding: 5px;
  justify-items: space-evenly;
}

.number{
  /* display: inline-block; */
  padding: 10% 10%;
  font-size:250%
}
.number:hover{
  background-color: greenyellow;
}


/* THIS IS FOR TEXT BOX ANSWER */
#textAnswer{
  /* display: grid;
  grid-template-columns: repeat(2,1fr);
  grid-gap: 10px; */
  width: 50%;
  /* margin-top: 2%;
  margin-left: 27.5%;
  margin-right: 37.5%; */
  margin: auto;
  padding: 5px;
}

#answerBox{
  width: 100%;
}

#submitBttn{
  text-align: center;
  margin: 1vh 1vh 0 1vh;
}

.submit {
  width: 100%;
  margin: auto;
}

.Location{
  font-size:200%;
}

#timer {
  font-size: 5vh;
}
.prog{
  width: 75%;
  padding-bottom: 4vh;
  white: red;
}

.game-synopsis-container {
  max-width: 100%;
  margin: 0 auto;
}

.back-button {
  font-size: 35px;
  color: black;
  text-align: left;
  padding: 30px 0px 10px 25px;
}

.game-title {
  background:black;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.synopsis-inner-container {
  max-width: 80%;
  margin: 0 auto;
}

.game-title h1 {
  font-size: 75px;
  color: black;
  text-decoration: underline;
}

.section-title h3 {
  font-size: 65px;
  margin-top: 25px;
  margin-bottom: 25px;
}

.synopsis p{
  font-size: 35px;
}

.instruction-questions p {
  font-size: 30px;
  text-align: center;
}

.instruction-questions h4 strong{
  font-size: 40px;
}

.instruction-questions ul {
  list-style-type: none;
}

.instruction-questions ul li {
  font-size: 40px;
  text-decoration: underline;
  
}
.instruction-gifs{
  width: 50%;
}

.section-divider hr {
  margin-top: 25px;
  margin-bottom: 25px;
  max-width: 550px;
  border-top: 5px solid rgb(255, 187, 0)
}

#notAtLocationIndicator{
  font-size: 35px;
}
