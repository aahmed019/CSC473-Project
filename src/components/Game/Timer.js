import React, { Component } from 'react';

class Timer extends Component {
    constructor(props) {
        super(props);
        this.state = {
            count: 1
        }
    }

    convertSeconds (seconds) {
        let min = Math.floor(seconds / 60);
        let sec = seconds % 60;
        return String(min).padStart(2, '0') + ':' + String(sec).padStart(2, '0');
    }

    render () {
        const { count } = this.state;
        
        return (
            <div id = "time">
                <h1>Time: { this.convertSeconds(count) }</h1>
            </div>
        )
    }

    componentDidMount () {
        const {startCount} = this.props;
        this.setState({
            count: startCount
        })
        this.myInterval = setInterval(() => {
            this.setState(prevState => ({
                count: prevState.count - 1
            }))
        }, 1000)
    }

    componentWillUnmount () {
        clearInterval(this.myInterval);
    }
}

export default Timer;