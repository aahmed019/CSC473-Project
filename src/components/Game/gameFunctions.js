import games from './games.json';
import React from 'react';
import ReactDOM from "react-dom";

// const render = Component =>
//     ReactDOM.render(Component, document.getElementById("root"));
// render(returnValue);


function getAnswer(gameIndex, questionIndex) {
    return function () {
        let userAnswer = document.getElementById("answer").value;
        let answer = games[gameIndex].answers[questionIndex];
        console.log(games[gameIndex].questions[questionIndex]);
        console.log(userAnswer);
        console.log(answer);
        if (userAnswer == answer) {
            document.getElementById("result").innerText = "Correct";
            console.log(questionIndex)
        }
        else {
            document.getElementById("result").innerText = "Wrong";
            console.log(questionIndex)
        }
    }
}

export default getAnswer