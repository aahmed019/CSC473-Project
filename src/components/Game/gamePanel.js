import React, { Component } from 'react';


class gamePanel extends Component {

    render() {
        const gameCardListStyle = {
            'list-style-type': 'none'
        }
        let listItems = this.props.games
            .map(item =>
                <div id={"game" + item.id} className="card">
                    <li className="card-body" key={item.id} >
                        <img
                            key={item.id}
                            className="card-img-top"
                            src={item.Thumbnail}
                            alt='' />
                        <h5 className="card-title">
                            <strong>{item.Title}</strong>
                        </h5>
                        <ul className="card-text">
                            <li key="rating">Rating: {item.Average_Rating} ({item.ReviewCount})</li>
                            <li key="location">Location: {item.Location}</li>
                            <li key="capacity"><strong>Capacity: {item.Capacity}</strong></li>
                            <li key="gameLength">Game Length: {item.Time_Limit / 60} minutes</li>
                            <li key="difficulty">Difficulty: {item.Difficulty}/5</li>
                        </ul>
                        {console.log(item.In_Progress)}
                        {
                            (item.In_Progess || item.Capacity === 0) ?
                                <button
                                    disabled={item.In_Progress}
                                    id={"bttn" + item.id}
                                    className="btn btn-lg btn-primary"
                                    type="button"
                                    value={item.id}>
                                    In Progress
                        </button> :
                                <button
                                    disabled={item.Capacity === 0}
                                    id={"bttn" + item.id}
                                    className="btn btn-lg btn-primary"
                                    type="button"
                                    onClick={this.props.func}
                                    value={item.id}>
                                    More Info
                        </button>
                        }
                    </li>
                </div>
            );
        return (
            <ul style={gameCardListStyle}>{listItems}</ul>
        )
    }

}
export default gamePanel;

//Props this file needs to run:
// id, thumbnail, title,location, capacity, timelimite, difficulty