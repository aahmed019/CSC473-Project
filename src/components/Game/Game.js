import React, { Component } from 'react';
import 'bootstrap/dist/css/bootstrap.css';
import games from './games.json';
// import getAnswer from './gameFunctions.js';
import './Game.css';
import Navbar from '../Navbar/Navbar';
import SideDrawerMenu from '../SideDrawerMenu/SideDrawerMenu';
import Backdrop from '../Backdrop/Backdrop';
import Footer from '../Footer/Footer';

// import puzzle component
import Puzzle from './Puzzle';


class Game extends Component {
  constructor(props) {
    super(props)
    this.state = {
      // game state
      gameState: true,
      // temp side drawer
      sideDrawerOpen: false
    }
  }

  /* this is temporary ****************************************** */
  drawerToggleClickHandler = () => {
    this.setState((prevState) => {
      return { sideDrawerMenuOpen: !prevState.sideDrawerMenuOpen };
    })
  };

  // clicking backdrop closes side drawer
  backdropClickHandler = () => {
    this.setState({ sideDrawerMenuOpen: false });
  }
  /* *************************************************************** */

  // position = async () => {
  //   await navigator.geolocation.getCurrentPosition(
  //     position => this.setState({
  //       latitude: position.coords.latitude,
  //       longitude: position.coords.longitude
  //     }), newState => console.log(newState))

  //   console.log(this.state.latitude, this.state.longitude)
  // }

  //Want to load the game in here based on the name
  render = () => {

    let backdrop;
    if (this.state.sideDrawerMenuOpen) {
      backdrop = <Backdrop click={this.backdropClickHandler} />
    }

    return (
      <div className="Game">

        <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.2/css/all.css"
          integrity="sha384-oS3vJWv+0UjzBfQzYUhtDYW+Pj2yciDJxpsK1OYPAYjqT085Qq/1cq5FLXAZQ7Ay"
          crossOrigin="anonymous" />
        <div>
          <Navbar drawerClickHandler={this.drawerToggleClickHandler} />
          <SideDrawerMenu show={this.state.sideDrawerMenuOpen} />
          {backdrop}
        </div>

        <div className = "game-state">
          <Puzzle />
        </div>
        <br />
        <Footer />
      </div>
    );
  }
}


export default Game;