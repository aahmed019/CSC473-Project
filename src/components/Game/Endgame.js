import React, { Component } from 'react';
import './Endgame.css';
import 'bootstrap/dist/css/bootstrap.css';
import { API, graphqlOperation } from 'aws-amplify';
import * as mutations from '../../graphql/mutations';
import * as queries from '../../graphql/queries';
//props this file needs:
// id, AtQuestion,Players

class Endgame extends Component {
    constructor(props) {
        super(props);
        this.handleSubmit = this.handleSubmit.bind(this);
    }
    // if this is the first time user wrintg review:
    // Create new review with id 
    // update reviewCount + 1 and calculate with that average
    // Else update current review of that id
    // query to get this user past review => + or - their rating / current reviewCount = averageRating
    async handleSubmit(e) {
        e.preventDefault();
        const userName = this.props.gameUserName;
        const userRating = Number(e.target.rating.value);
        const userReview = e.target.review.value;
        const reviewId = "00" + this.props.gameId + userName;
        const reviewCount = this.props.gameReviewCount;

        console.log('Rating: ' + userRating);
        console.log('Review: ' + userReview);
        console.log('')
        const newReview = {
            id: reviewId,
            rating: userRating,
            review: userReview,
            username: userName,
            reviewGameId: this.props.gameId
        }

        try {
            // Handles the review information, updates the rating based on the average of all reviews
            const newReviewCount = {
                id: this.props.gameId,
                ReviewCount: reviewCount + 1,
                Average_Rating: (Math.round((Number(this.props.gameAverageRating) + userRating) / (reviewCount + 1) * 100) / 100).toString()
            }

            await API.graphql(graphqlOperation(mutations.createReview, { input: newReview }));
            alert("Your review is submitted!")
            try {
                await API.graphql(graphqlOperation(mutations.updateGame, { input: newReviewCount }));
                alert("Your review is submitted!")
            } catch (errors) { console.log(errors) }

        } catch (errors) {
            try {
                const exReview = await API.graphql(graphqlOperation(queries.getReview, { id: reviewId }));
                console.log(exReview.data.getReview);
                let localReview = exReview.data.getReview;
                let localRating = Number(localReview.rating);
                let changedRating = Number(userRating - localRating);
                const oldReviewCount = {
                    id: this.props.gameId,
                    ReviewCount: reviewCount,
                    Average_Rating: (Math.round((Number(this.props.gameAverageRating) + changedRating) / (reviewCount + 1) * 100) / 100).toString()
                }
                try {
                    await API.graphql(graphqlOperation(mutations.updateGame, { input: oldReviewCount }));
                    alert("Your review is submitted!")
                } catch (errors) { console.log(errors) }

                await API.graphql(graphqlOperation(mutations.updateReview, { input: newReview }));
                alert("Your review is submitted!")
            } catch (error) { console.log(error) }
        }
    }
    async componentDidMount() {
        console.log("Endgame Component Did Mount");
        const nQuestion = {
            id: this.props.gameId,
            Finished: false
        }
        try {
            await API.graphql(graphqlOperation(mutations.updateGame, { input: nQuestion }));
        } catch (errors) { console.log("Errors on reset game: ", errors) };
    }

    render() {
        //User winning page
        if (this.props.outcome) {
            return (
                <div className="ending-page" >
                    <p id="message"><strong>WINNER WINNER CHICKEN DINNER</strong></p>
                    <img id="outcome-img" src="https://res.cloudinary.com/teepublic/image/private/s--xRlv7eQ_--/t_Preview/b_rgb:42332c,c_limit,f_jpg,h_630,q_90,w_630/v1541876140/production/designs/3479222_0.jpg" alt="win" />

                    <div className='section-divider'>
                        <hr />
                    </div>
                    <div className='section-title'>
                        <h3>Tell us about your experience</h3>
                    </div>
                    <div className="review-form-container">
                        <form id="review-form" onSubmit={this.handleSubmit}>
                            <div className="form-group" id="rating">
                                <label for="rating" className="form-label">Rating (Required)</label>
                                <select className="form-control" id="rating" required>
                                    <option defaultValue>1</option>
                                    <option>2</option>
                                    <option>3</option>
                                    <option>4</option>
                                    <option>5</option>
                                </select>
                            </div>
                            <br />
                            <div className="form-group">
                                <label for="feedback" className="form-label">Review (Required)</label>
                                <textarea className="form-control" id="review" rows="5" required />
                            </div>
                            <br />
                            <button className="review-submit-button btn-lg" type="submit">
                                Submit
                                {/* <a href="/Game">
                                    Submit
                                </a> */}
                            </button>
                        </form>
                    </div>
                </div>

            )
        }
        // User losing page
        else {
            return (
                <div className="ending-page">
                    <p id="message"><strong>GAME OVER</strong></p>
                    <img id="outcome-img" src="https://fsa.zobj.net/crop.php?r=0ZfAZ3qrrkdCcPeZC4mS_4Cc8RI-uhrv1IL99GWh5jzxRIDm2jead7tvMb0_hHbtbh6Qv9pymBVux2TFQZyVayeU5Zn73C1yGxpGA0ls96ff7pcy1H-jHU03lZhSkqpQ5RkUoM6Ip9NuXR2Y" alt="lose" />
                    <div className='section-divider'>
                        <p>_________________________________</p>
                    </div>
                    <div className='section-title'>
                        <h3>Tell us about your experience</h3>
                    </div>
                    <div className="review-form-container">
                        <form id="review-form" onSubmit={this.handleSubmit}>
                            <div className="form-group" id="rating">
                                <label for="rating" className="form-label">Rating (Required)</label>
                                <select className="form-control" id="rating-select" required>
                                    <option defaultValue>1</option>
                                    <option>2</option>
                                    <option>3</option>
                                    <option>4</option>
                                    <option>5</option>
                                </select>
                            </div>
                            <br />
                            <div className="form-group">
                                <label for="feedback" className="form-label">Review (Required)</label>
                                <textarea className="form-control" id="feedback" rows="5" required />
                            </div>
                            <br />
                            <button className="review-submit-button btn-lg" type="submit">
                                Submit
                                {/* <a href="/Game">
                                    Submit
                                </a> */}
                            </button>
                        </form>
                    </div>
                </div>
            )
        }

    }

}



export default Endgame;