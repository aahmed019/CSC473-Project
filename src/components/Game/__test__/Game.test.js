import React from 'react';
import ReactDOM from 'react-dom';
import Game from './../Game';
import Timer from './../Timer';
import { shallow, mount } from 'enzyme';


describe("Game component testing", () => {
    // crash test
    it("renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<Game />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    // test getGameID
    describe("component renders with only one game div", () => {
        // let wrapper;
        // beforeEach(() => { wrapper = shallow(<Game />); });
        let wrapper = mount(<Game />);

        it('includes a div with className game', () => {
            expect(wrapper.find(Game).length).toBe(1);
        });

    })

    describe("component renders different pages based on states", () => {
        let wrapper;
        beforeEach(() => { wrapper = mount(<Game />); });
        
        // renders game list
        it('user starts at game list page', () => {
            expect(wrapper.find('div.game-list').length).toBe(1);
        });

        // renders synopsis
        it('renders div with className synopsis', () => {
            // wrapper.setState( {
            //     gameReady: true,
            //     gameSynopsis: 1,
            //     gameStart: 0,
            // });
            //wrapper.update();
            const button = wrapper.find('#bttn0');
            expect(button.length).toBe(1);
            button.simulate('click');
            expect(wrapper.find('div.synopsis').length).toBe(1);
        });

        it('renders div with className gameInterface', () => {
            // wrapper.setState ( {
            //     gameReady: true,
            //     gameSynopsis: 0,
            //     gameStart: 1,
            // });
            // wrapper.update();
            let button = wrapper.find('#bttn0');
            expect(button.length).toBe(1);
            button.simulate('click');
            button = wrapper.find('#start-btn');
            button.simulate('click');
            expect(wrapper.find('div.gameInterface').length).toBe(1);
            expect(wrapper.find(Timer).length).toBe(1);
        })
    })

    describe('go back from each page via exit button', () => {
        let wrapper;
        beforeEach(() => { wrapper = mount(<Game />); });

        it('return to home page from synopsis', () => {
            let button = wrapper.find('#bttn0');
            expect(button.length).toBe(1);
            button.simulate('click');
            button = wrapper.find('button.btn-danger');
            expect(button.length).toBe(1);
            button.simulate('click');
            expect(wrapper.find('div.body-page'));
        });

        it('returns to synopsis page from game page', () => {
            let button = wrapper.find('#bttn0');
            expect(button.length).toBe(1);
            button.simulate('click');
            button = wrapper.find('button.btn-danger');
            expect(button.length).toBe(1);
            button.simulate('click');
            expect(wrapper.find('div.gameInterface'));
        })
    });
})