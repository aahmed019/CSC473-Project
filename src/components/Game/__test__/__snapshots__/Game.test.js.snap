// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game component testing snapshot testing game list page matches the snapshot 1`] = `
<div
  className="Game"
>
  <br />
  <p
    className="Location"
  >
    Click the button to get your coordinates.
  </p>
  <p
    className="Location"
  >
     
  </p>
  <button
    className="Location"
    onClick={[Function]}
  >
    Location
  </button>
  <br />
  <link
    crossOrigin="anonymous"
    href="https://use.fontawesome.com/releases/v5.8.2/css/all.css"
    integrity="sha384-oS3vJWv+0UjzBfQzYUhtDYW+Pj2yciDJxpsK1OYPAYjqT085Qq/1cq5FLXAZQ7Ay"
    rel="stylesheet"
  />
  <br />
  <div
    className="exit"
  >
    <button
      className="btn btn-lg btn-danger"
      type="button"
    >
      <a
        href="/"
      >
          Exit  
      </a>
    </button>
  </div>
  <div
    className="game-list"
  >
    <div
      className="card"
    >
      <ul>
        <li
          className="card-body"
        >
          <img
            alt=""
            className="card-img-top"
            src="https://i.ibb.co/sVKnY3R/74419834-2503997649921511-7175701397294284800-n.jpg"
          />
          <h5
            className="card-title"
          >
            <strong>
              The Curious Case of Room STC01
            </strong>
          </h5>
          <ul
            className="card-text"
          >
            <li>
              Location:
            </li>
            <li>
              Capacity:
            </li>
            <li>
              Game Length:
            </li>
            <li>
              Difficulty: 
              2
              /5
            </li>
          </ul>
          <button
            className="btn btn-lg btn-primary"
            id="bttn0"
            onClick={[Function]}
            type="button"
            value={0}
          >
            START
          </button>
        </li>
        <li
          className="card-body"
        >
          <img
            alt=""
            className="card-img-top"
            src="https://cdn.discordapp.com/attachments/629102063107964961/644893758793449494/image0.jpg"
          />
          <h5
            className="card-title"
          >
            <strong>
              Hero of the Angle
            </strong>
          </h5>
          <ul
            className="card-text"
          >
            <li>
              Location:
            </li>
            <li>
              Capacity:
            </li>
            <li>
              Game Length:
            </li>
            <li>
              Difficulty: 
              4
              /5
            </li>
          </ul>
          <button
            className="btn btn-lg btn-primary"
            id="bttn1"
            onClick={[Function]}
            type="button"
            value={1}
          >
            START
          </button>
        </li>
        <li
          className="card-body"
        >
          <img
            alt=""
            className="card-img-top"
            src="https://picsum.photos/200/300?random=1"
          />
          <h5
            className="card-title"
          >
            <strong>
              Wingate Hall
            </strong>
          </h5>
          <ul
            className="card-text"
          >
            <li>
              Location:
            </li>
            <li>
              Capacity:
            </li>
            <li>
              Game Length:
            </li>
            <li>
              Difficulty: 
              3
              /5
            </li>
          </ul>
          <button
            className="btn btn-lg btn-primary"
            id="bttn2"
            onClick={[Function]}
            type="button"
            value={2}
          >
            START
          </button>
        </li>
        <li
          className="card-body"
        >
          <img
            alt=""
            className="card-img-top"
            src="https://picsum.photos/200/300?random=1"
          />
          <h5
            className="card-title"
          >
            <strong>
              Marshak Hall
            </strong>
          </h5>
          <ul
            className="card-text"
          >
            <li>
              Location:
            </li>
            <li>
              Capacity:
            </li>
            <li>
              Game Length:
            </li>
            <li>
              Difficulty: 
              4
              /5
            </li>
          </ul>
          <button
            className="btn btn-lg btn-primary"
            id="bttn3"
            onClick={[Function]}
            type="button"
            value={3}
          >
            START
          </button>
        </li>
        <li
          className="card-body"
        >
          <img
            alt=""
            className="card-img-top"
            src="https://cdn.discordapp.com/attachments/629102063107964961/644893758793449494/image0.jpg"
          />
          <h5
            className="card-title"
          >
            <strong>
              Hero of the Angle
            </strong>
          </h5>
          <ul
            className="card-text"
          >
            <li>
              Location:
            </li>
            <li>
              Capacity:
            </li>
            <li>
              Game Length:
            </li>
            <li>
              Difficulty: 
              4
              /5
            </li>
          </ul>
          <button
            className="btn btn-lg btn-primary"
            id="bttn4"
            onClick={[Function]}
            type="button"
            value={4}
          >
            START
          </button>
        </li>
      </ul>
    </div>
  </div>
  <br />
</div>
`;
